Library/Homebrew/utils/github.rb

```diff
@@ -371,6 +372,28 @@ def self.get_artifact_url(workflow_array)
     artifact.last["archive_download_url"]
   end
 
+  def self.download_artifact(url, artifact_id, dir = Pathname.pwd)
+    odie "Credentials must be set to access the Artifacts API" if API.credentials_type == :none
+
+    token = API.credentials
+    curl_args = ["--header", "Authorization: token #{token}"]
+
+    # Download the artifact as a zip file and unpack it into `dir`. This is
+    # preferred over system `curl` and `tar` as this leverages the Homebrew
+    # cache to avoid repeated downloads of (possibly large) bottles.
+    FileUtils.chdir dir do
```

Comment:
```suggestion
    dir.cd do
```

if it's a `Pathname`

Reply:
Not always, unfortunately, since it uses `dir` from here

https://github.com/Homebrew/brew/blob/111f50492585a58324563107453ac7e043351157/Library/Homebrew/dev-cmd/pr-pull.rb#L468

Reply:
Ah ok, np.

Reply:
This is actually one of only two instances where `FileUtils.chdir` is used instead of `Dir.chdir`.

Reply:
> This is actually one of only two instances where `FileUtils.chdir` is used instead of `Dir.chdir`.

Actually, looking at the only existing caller, the `chdir` call might not even be needed.

> Nit: Maybe it'd be good to add a type signature for this function since it's not obvious what it takes as parameters. Let Sorbet do a little work as well.

Sure, let me add a signature.